### PR TITLE
Refactor bed sql

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Refactor bed sql

### Added

- Tests

### Changed

- Refactor bed sql queries and filters

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b refactor_bed_sql -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [x] Tests executed by HS
- [x] "Merge and deploy" approved by @Vince-janv @seallard @ChrOertlin 
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/server/api.py [hunk 1]
```diff
@@ -357,7 +357,7 @@ def options():
             for organism in db.organisms()
```
Comment (ChrOertlin) [inline cg/server/api.py:357]: Should we add a pydantic module for organisms? It is currently a dict

File: cg/server/api.py [hunk 2]
```diff
@@ -323,27 +323,27 @@ def analyses():
...
+    customers: Optional[Customer] = (
         db.Customer.query.all() if g.current_user.is_admin else g.current_user.customers
     )
 
-    apptag_groups = {"ext": []}
-    for application_obj in db.applications(archived=False):
-        if not application_obj.versions:
-            LOG.debug("Skipping application %s that doesn't have a price", application)
+    apptag_groups: Dict[str, list[str]] = {"ext": []}
+    for application in db.applications(archived=False):
```
Comment (ChrOertlin) [inline cg/server/api.py:331]: Does application require a type hint here?

**Review by ChrOertlin — APPROVED** [review]

File: cg/server/api.py (hunk 2 above)
Comment (henrikstranneheim) [reply cg/server/api.py:331]: Python does not support type hints in for loop initialization. Then one have  to add it before the loop. We can fix this when doing the application refactoring

File: cg/server/api.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg/server/api.py:357]: Good suggestion, but let's do it when we refactor the organism SQL code. I am trying to not extend the scope outside Bed refactoring too much

File: cg/store/status_bed_filters.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        "get_beds_by_name": get_beds_by_name,
+        "get_not_archived_beds": get_not_archived_beds,
+        "order_beds_by_name": order_beds_by_name,
+    }
+    for function in functions:
+        beds: Query = filter_map[function](
+            beds=beds,
+            bed_name=bed_name,
+        )
+    return beds
```
Comment (Vince-janv) [inline cg/store/status_bed_filters.py:39]: This was what i envisioned when I asked about it in during the BB-seminar. I feels much more reliable than strings that map to functions, but requires importing the function on use.
```suggestion
def apply_bed_filter(
    beds: Query,
    functions: List[Callable],
    bed_name: Optional[str] = None,
) -> Query:
    """Apply filtering functions and return filtered results."""
    for function in functions:
        beds: Query = function(
            beds=beds,
            bed_name=bed_name,
        )
    return beds
```

And then on usage:
```
apply_bed_filter(beds=self._get_bed_query(),[get_beds_by_name, get_not_archived_beds])
```

File: cg/server/api.py [hunk 4]
```diff
@@ -157,7 +157,7 @@ def families():
         records = db.cases_to_mip_analyze()
         count = len(records)
     else:
-        customer_objs: Optional[List[models.Customer]] = (
+        customer_objs: Optional[List[Customer]] = (
```
Comment (Vince-janv) [inline cg/server/api.py:160]:
```suggestion
        customers: Optional[List[Customer]] = (
```

File: cg/server/api.py [hunk 5]
```diff
@@ -224,7 +224,7 @@ def samples():
     elif request.args.get("status") == "sequencing":
         samples_q = db.samples_to_sequence()
     else:
-        customer_objs: Optional[models.Customer] = (
+        customer_objs: Optional[Customer] = (
```
Comment (Vince-janv) [inline cg/server/api.py:227]:
```suggestion
        customer: Optional[Customer] = (
```

File: cg/server/api.py [hunk 6]
```diff
@@ -271,7 +271,7 @@ def sample_in_collaboration(sample_id):
 @BLUEPRINT.route("/pools")
 def pools():
     """Fetch pools."""
-    customer_objs: Optional[models.Customer] = (
+    customer_objs: Optional[Customer] = (
```
Comment (Vince-janv) [inline cg/server/api.py:274]:
```suggestion
    customers: Optional[Customer] = (
```

**Review by Vince-janv — APPROVED** [review]
Nice, step by step! 💪

File: cg/server/api.py (hunk 5 above)
Comment (henrikstranneheim) [reply cg/server/api.py:229]: This one looks a  bit mixed in the code between a `Optional[List[Customer]]` and Optional[Customer] and I dot want to dive into refactoring this part now. Let's do it when refactoring Customer SQL

File: cg/server/api.py (hunk 6 above)
Comment (henrikstranneheim) [reply cg/server/api.py:276]: Same here

File: cg/store/status_bed_filters.py (hunk 3 above)
Comment (henrikstranneheim) [reply cg/store/status_bed_filters.py]: Very nice suggestion! Made a version of this using a class to make it easier to get all available filters and make it easy to extend/refactor in the future. Have a look

File: cg/server/api.py (hunk 4 above)
Comment (henrikstranneheim) [reply cg/server/api.py]: Done

File: cg/store/status_bed_filters.py (hunk 3 above)
Comment (Vince-janv) [reply cg/store/status_bed_filters.py]: Good solution 🚀

**Review by seallard — APPROVED** [review]

**Comment by henrikstranneheim** [comment]
##  Rest-api loads and is browsable ✅ 

Screenshot 2023-02-28 at 15 03 27

**Comment by henrikstranneheim** [comment]
I do not think these code smells are relevant - I will ignore them for now

**Comment by henrikstranneheim** [comment]
Deployed on Hasta:
```
θ83° [henrik.stranneheim@hasta:~] [P_base] $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-prod.sh -e P_main -t cg -b master -a
/home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))

Collecting git+https://github.com/Clinical-Genomics/cg@master
  Running command git clone --quiet https://github.com/Clinical-Genomics/cg /home/henrik.stranneheim/tmp/pip-req-build-0kwuypak
  Cloning https://github.com/Clinical-Genomics/cg (to revision master) to ./tmp/pip-req-build-0kwuypak
  Resolved https://github.com/Clinical-Genomics/cg to commit ef1516362b4ec2feb0ab56b90e09b9ad3c46b181
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: cg
  Building wheel for cg (pyproject.toml): started
  Building wheel for cg (pyproject.toml): finished with status 'done'
  Created wheel for cg: filename=cg-26.3.5-py3-none-any.whl size=672899 sha256=5e5854c9cb380ec0fd7e054eda4e02143a8cfe7a9046f200cf40c5dca17290c4
  Stored in directory: /home/henrik.stranneheim/tmp/pip-ephem-wheel-cache-fovtm1_6/wheels/4b/55/66/90c6ed137f9153944f27209bd5c3790044b15cf24d2c7f573b
Successfully built cg
Installing collected packages: cg
  Attempting uninstall: cg
    Found existing installation: cg 26.3.4
    Uninstalling cg-26.3.4:
      Successfully uninstalled cg-26.3.4
Successfully installed cg-26.3.5
Collecting git+https://github.com/Clinical-Genomics/cg@master
  Running command git clone --quiet https://github.com/Clinical-Genomics/cg /home/henrik.stranneheim/tmp/pip-req-build-y5po8m4e
  Cloning https://github.com/Clinical-Genomics/cg (to revision master) to ./tmp/pip-req-build-y5po8m4e
  Resolved https://github.com/Clinical-Genomics/cg to commit ef1516362b4ec2feb0ab56b90e09b9ad3c46b181
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Requirement already satisfied: housekeeper>=2.5 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (3.2.0)
Requirement already satisfied: python-dateutil in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (2.8.2)
Requirement already satisfied: typing-extensions in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (4.4.0)
Requirement already satisfied: alembic in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.8.1)
Requirement already satisfied: google-auth in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (2.13.0)
Requirement already satisfied: genologics in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (0.4.6)
Requirement already satisfied: setuptools>=39.2.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (65.5.0)
Requirement already satisfied: petname in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (2.6)
Requirement already satisfied: coloredlogs in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (15.0.1)
Requirement already satisfied: Click<7.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (6.7)
Requirement already satisfied: gunicorn in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (20.1.0)
Requirement already satisfied: markupsafe<2.1 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (2.0.1)
Requirement already satisfied: wtforms<3.0.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (2.3.3)
Requirement already satisfied: blinker in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.5)
Requirement already satisfied: Flask-Alchy in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (0.5.0)
Requirement already satisfied: Alchy in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (2.2.2)
Requirement already satisfied: Flask>=1.1.1 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.1.4)
Requirement already satisfied: Flask-CORS in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (3.0.10)
Requirement already satisfied: Flask-Admin in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.6.0)
Requirement already satisfied: pyyaml in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (6.0)
Requirement already satisfied: flask-wtf in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.1.1)
Requirement already satisfied: requests-cache in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (0.9.6)
Requirement already satisfied: requests[security] in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (2.28.1)
Requirement already satisfied: cgmodels>=0.10.5 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (0.10.5)
Requirement already satisfied: Flask-SQLAlchemy==2.1 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (2.1)
Requirement already satisfied: paramiko in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (2.11.0)
Requirement already satisfied: packaging in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (21.3)
Requirement already satisfied: sendmail-container in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (0.2.0)
Requirement already satisfied: marshmallow>3 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (3.18.0)
Requirement already satisfied: SQLAlchemy<1.4 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.3.24)
Requirement already satisfied: pymysql in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.0.2)
Requirement already satisfied: lxml in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (4.9.1)
Requirement already satisfied: werkzeug<1.0.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (0.16.1)
Requirement already satisfied: pandas in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.3.5)
Requirement already satisfied: tabulate in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (0.9.0)
Requirement already satisfied: pydantic in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.10.2)
Requirement already satisfied: Flask-Dance in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (5.1.0)
Requirement already satisfied: ansi in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (0.3.6)
Requirement already satisfied: email-validator in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (1.3.0)
Requirement already satisfied: openpyxl==3.0.10 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cg==26.3.5) (3.0.10)
Requirement already satisfied: et-xmlfile in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from openpyxl==3.0.10->cg==26.3.5) (1.1.0)
Requirement already satisfied: atomicwrites in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (1.4.1)
Requirement already satisfied: colorama in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (0.4.5)
Requirement already satisfied: attrs in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (22.1.0)
Requirement already satisfied: coverage in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (6.5.0)
Requirement already satisfied: pytest in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (7.1.3)
Requirement already satisfied: pytest-cov in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (4.0.0)
Requirement already satisfied: zipp in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (3.9.0)
Requirement already satisfied: more-itertools in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (8.14.0)
Requirement already satisfied: importlib-metadata in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (5.0.0)
Requirement already satisfied: py in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (1.11.0)
Requirement already satisfied: dataclasses in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (0.6)
Requirement already satisfied: wcwidth in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (0.2.5)
Requirement already satisfied: pyparsing in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (2.4.7)
Requirement already satisfied: pluggy in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.5->cg==26.3.5) (1.0.0)
Requirement already satisfied: Jinja2<3.0,>=2.10.1 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from Flask>=1.1.1->cg==26.3.5) (2.11.3)
Requirement already satisfied: itsdangerous<2.0,>=0.24 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from Flask>=1.1.1->cg==26.3.5) (1.1.0)
Requirement already satisfied: rich in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from housekeeper>=2.5->cg==26.3.5) (12.6.0)
Requirement already satisfied: ruamel.yaml in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from housekeeper>=2.5->cg==26.3.5) (0.17.21)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from Alchy->cg==26.3.5) (5.1.1)
Requirement already satisfied: Mako in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from alembic->cg==26.3.5) (1.2.3)
Requirement already satisfied: importlib-resources in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from alembic->cg==26.3.5) (5.10.0)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from coloredlogs->cg==26.3.5) (10.0)
Requirement already satisfied: idna>=2.0.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from email-validator->cg==26.3.5) (3.4)
Requirement already satisfied: dnspython>=1.15.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from email-validator->cg==26.3.5) (2.2.1)
Requirement already satisfied: Six in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from Flask-CORS->cg==26.3.5) (1.16.0)
Requirement already satisfied: oauthlib in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from Flask-Dance->cg==26.3.5) (3.2.2)
Requirement already satisfied: urlobject in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from Flask-Dance->cg==26.3.5) (2.4.3)
Requirement already satisfied: requests-oauthlib>=1.0.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from Flask-Dance->cg==26.3.5) (1.3.1)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from google-auth->cg==26.3.5) (5.2.0)
Requirement already satisfied: rsa<5,>=3.1.4 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from google-auth->cg==26.3.5) (4.9)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from google-auth->cg==26.3.5) (0.2.8)
Requirement already satisfied: numpy>=1.17.3 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from pandas->cg==26.3.5) (1.21.6)
Requirement already satisfied: pytz>=2017.3 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from pandas->cg==26.3.5) (2022.4)
Requirement already satisfied: bcrypt>=3.1.3 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from paramiko->cg==26.3.5) (4.0.1)
Requirement already satisfied: cryptography>=2.5 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from paramiko->cg==26.3.5) (38.0.1)
Requirement already satisfied: pynacl>=1.0.1 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from paramiko->cg==26.3.5) (1.5.0)
Requirement already satisfied: appdirs>=1.4.4 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from requests-cache->cg==26.3.5) (1.4.4)
Requirement already satisfied: url-normalize>=1.4 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from requests-cache->cg==26.3.5) (1.4.3)
Requirement already satisfied: cattrs>=1.8 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from requests-cache->cg==26.3.5) (22.2.0)
Requirement already satisfied: urllib3>=1.25.5 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from requests-cache->cg==26.3.5) (1.26.12)
Requirement already satisfied: charset-normalizer<3,>=2 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from requests[security]->cg==26.3.5) (2.1.1)
Requirement already satisfied: certifi>=2017.4.17 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from requests[security]->cg==26.3.5) (2022.9.24)
Requirement already satisfied: fastapi in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from sendmail-container->cg==26.3.5) (0.85.1)
Requirement already satisfied: uvicorn in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from sendmail-container->cg==26.3.5) (0.5.1)
Requirement already satisfied: python-multipart in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from sendmail-container->cg==26.3.5) (0.0.5)
Requirement already satisfied: exceptiongroup in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cattrs>=1.8->requests-cache->cg==26.3.5) (1.0.0rc9)
Requirement already satisfied: cffi>=1.12 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cryptography>=2.5->paramiko->cg==26.3.5) (1.15.1)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from pyasn1-modules>=0.2.1->google-auth->cg==26.3.5) (0.4.8)
Requirement already satisfied: starlette==0.20.4 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from fastapi->sendmail-container->cg==26.3.5) (0.20.4)
Requirement already satisfied: anyio<5,>=3.4.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from starlette==0.20.4->fastapi->sendmail-container->cg==26.3.5) (3.6.1)
Requirement already satisfied: python-dotenv>=0.10.4 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from pydantic->cg==26.3.5) (0.21.0)
Requirement already satisfied: iniconfig in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from pytest->cgmodels>=0.10.5->cg==26.3.5) (1.1.1)
Requirement already satisfied: tomli>=1.0.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from pytest->cgmodels>=0.10.5->cg==26.3.5) (2.0.1)
Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from rich->housekeeper>=2.5->cg==26.3.5) (0.9.1)
Requirement already satisfied: pygments<3.0.0,>=2.6.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from rich->housekeeper>=2.5->cg==26.3.5) (2.13.0)
Requirement already satisfied: ruamel.yaml.clib>=0.2.6 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from ruamel.yaml->housekeeper>=2.5->cg==26.3.5) (0.2.6)
Requirement already satisfied: uvloop in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from uvicorn->sendmail-container->cg==26.3.5) (0.17.0)
Requirement already satisfied: h11 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from uvicorn->sendmail-container->cg==26.3.5) (0.14.0)
Requirement already satisfied: httptools in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from uvicorn->sendmail-container->cg==26.3.5) (0.5.0)
Requirement already satisfied: websockets>=6.0 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from uvicorn->sendmail-container->cg==26.3.5) (10.3)
Requirement already satisfied: pycparser in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from cffi>=1.12->cryptography>=2.5->paramiko->cg==26.3.5) (2.21)
Requirement already satisfied: sniffio>=1.1 in /home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages (from anyio<5,>=3.4.0->starlette==0.20.4->fastapi->sendmail-container->cg==26.3.5) (1.3.0)
/home/proj/production/bin/miniconda3/envs/P_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))

remote: Enumerating objects: 521, done.
remote: Counting objects: 100% (340/340), done.
remote: Compressing objects: 100% (116/116), done.
remote: Total 225 (delta 183), reused 142 (delta 108), pack-reused 0
Receiving objects: 100% (225/225), 37.62 KiB | 0 bytes/s, done.
Resolving deltas: 100% (183/183), completed with 71 local objects.
From https://github.com/Clinical-Genomics/cg
   e4c1ecf..ef15163  master     -> origin/master
 * [new branch]      add_file_tags_to_protect_in_Housekeeper_for_balsamic_analysis -> origin/add_file_tags_to_protect_in_Housekeeper_for_balsamic_analysis
   2d58faa..83e452c  delivery-fastq -> origin/delivery-fastq
   f9a2abb..1ec843c  refactor_invoice_sql -> origin/refactor_invoice_sql
   f211eea..d912a8e  restrict-cors -> origin/restrict-cors
 * [new branch]      rnafusion_params_file -> origin/rnafusion_params_file
 * [new tag]         v26.3.5    -> v26.3.5
Already on 'master'
Your branch is behind 'origin/master' by 2 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Updating e4c1ecf..ef15163
Fast-forward
 .bumpversion.cfg                              |   2 +-
 cg/__init__.py                                |   2 +-
 cg/apps/gens.py                               |   4 ++--
 cg/constants/constants.py                     |   1 +
 cg/server/api.py                              |  42 +++++++++++++++++++++---------------------
 cg/store/api/findbasicdata.py                 |  44 ++++++++++++++++++++++++++++----------------
 cg/store/status_bed_filters.py                |  41 +++++++++++++++++++++++++++++++++++++++++
 setup.py                                      |   2 +-
 tests/cli/compress/test_cli_compress_fastq.py |   5 +++--
 tests/conftest.py                             |  47 ++++++++++++++++++++++++++++-------------------
 tests/store/api/__init__.py                   |   0
 tests/store/api/test_findbasicdata.py         | 101 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 tests/store/api/test_findbusinessdata.py      |   3 +--
 tests/store/conftest.py                       |   8 ++++----
 tests/store/test_status_bed_filters.py        |  47 +++++++++++++++++++++++++++++++++++++++++++++++
 tests/store_helpers.py                        |  22 ++++++++++++----------
 16 files changed, 292 insertions(+), 79 deletions(-)
 create mode 100644 cg/store/status_bed_filters.py
 create mode 100644 tests/store/api/__init__.py
 create mode 100644 tests/store/api/test_findbasicdata.py
 create mode 100644 tests/store/test_status_bed_filters.py
/home/henrik.stranneheim
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/henrik.stranneheim
done.
Log deploy... done.
cg, version 26.3.5
```

**Pull request merged**